### PR TITLE
Add beaconing support (heartbeats) to  core and agent

### PR DESCRIPTION
**Pull request by GrayTShirt**
I am going to squash the commits in this pull request before approval, but in the mean time here it is...

issue #5

### CODE REVIEW

File: src/agent.c [hunk 1]
```diff
@@ -318,7 +420,7 @@ int main(int argc, char **argv)
 	}
 	logger(LOG_NOTICE, "starting up");
 
-	logger(LOG_DEBUG, "allocating 0MQ context");
+	logger(LOG_DEBUG, "allocating 0MQ contexts");
```
Comment (filefrog) [inline src/agent.c]: No longer plural

File: src/agent.c [hunk 2]
```diff
@@ -411,4 +551,4 @@ int main(int argc, char **argv)
 		}
 	}
 	return 0;
-}
+} /* }}} */
```
Comment (filefrog) [inline src/agent.c]: Can you remove all of these fold markers, or move them to a different commit / PR?

File: src/config.c [hunk 3]
```diff
@@ -21,36 +21,38 @@
...
+#define T_KEYWORD_USER         0x0b
+#define T_KEYWORD_GROUP        0x0c
+#define T_KEYWORD_BROADCAST    0x0d
+#define T_KEYWORD_WINDOW       0x0e
+#define T_KEYWORD_COUNTER      0x0f
+#define T_KEYWORD_SAMPLE       0x10
+#define T_KEYWORD_NSCAPORT     0x11
+#define T_KEYWORD_KEYSFILE     0x12
+#define T_KEYWORD_MAXEVENTS    0x13
+#define T_KEYWORD_RATE         0x14
```
Comment (filefrog) [inline src/config.c]: This belongs in a cleanup commit / PR

File: src/config.c [hunk 4]
```diff
@@ -740,12 +751,13 @@ int deconfigure(server_t *s)
 
 	free(s->config.listener);     s->config.listener     = NULL;
 	free(s->config.controller);   s->config.controller   = NULL;
-	free(s->config.broadcast);    s->config.broadcast   = NULL;
+	free(s->config.broadcast);    s->config.broadcast    = NULL;
```
Comment (filefrog) [inline src/config.c]: This belongs in a separate cleanup commit / PR

File: src/core.c [hunk 5]
```diff
@@ -905,6 +924,11 @@ int core_kernel_thread(void *zmq, server_t *server) /* {{{ */
 	kernel_t *kernel = vmalloc(sizeof(kernel_t));
 	kernel->server = server;
 
+	kernel->sweep.interval     = server->interval.sweep;
+	/* kernel->tick.interval      = server->interval.tick;
+	kernel->freshness.interval = server->interval.freshness;
+	kernel->savestate.interval = server->interval.savestate; */
+
```
Comment (filefrog) [inline src/core.c]: Remove the added-but-commented-out code

File: t/broadcast [hunk 6]
```diff
@@ -9,15 +9,19 @@ done
 LISTENER="ipc://${ROOT}/bolo.listener.sock"
 CONTROLLER="ipc://${ROOT}/bolo.controller.sock"
 BROADCAST="ipc://${ROOT}/bolo.broadcast.sock"
+BEACON="ipc://${ROOT}/bolo.beacon.sock"
```
Comment (filefrog) [inline t/broadcast]: Does the t/broadcast test benefit from beaconing support?

File: t/core [hunk 7]
```diff
@@ -9,14 +9,18 @@ done
 LISTENER="ipc://${ROOT}/bolo.listener.sock"
 CONTROLLER="ipc://${ROOT}/bolo.controller.sock"
 BROADCAST="ipc://${ROOT}/bolo.broadcast.sock"
+BEACON="ipc://${ROOT}/bolo.beacon.sock"
```
Comment (filefrog) [inline t/core]: Does t/core benefit from a beacon config?

File: t/lib [hunk 8]
```diff
@@ -158,3 +158,24 @@ string_is() {
...
   tdiag "ok ${msg}"
 }
+
+###############################################################################
+#
+# string_like - assert that two strings are similar via regex
+# USAGE: string_like ${actual} ${expected} \
+#                  "message to print on failure"
+#
+string_like() {
```
Comment (filefrog) [inline t/lib]: :metal: this is AWESOME

**Comment by filefrog** [comment]
Need to get this PR down to just stuff that matters.  Drop the extra whitespace changes and the line markers (submit them separately).

Also, the documentation reads a bit woodenly.  If you want me to, I can patch it up, post-merge.

**Comment by GrayTShirt** [comment]
Ok so I manually reverted the cleanup changes, but I'm having trouble squashing commits

**Comment by filefrog** [comment]
What sort of trouble?

**Comment by filefrog** [comment]
Also, we need to talk about Copyright.  I should be on IRC #bolo all day today
